cluster-service/deploy/templates/azure-operators-managed-identities-config.configmap.yaml

```diff
@@ -77,7 +77,7 @@ data:
           - resourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .id }}'
             name: '{{ .name }}'
         {{- end }}
-        optional: true
+        optional: false
```

Comment:
This is not correct. It is still an optional identity. It is only required when etcd data level encryption with kms is enabled. We refer to MIs that become mandatory when their corresponding functionality is enabled  as "optional: true" in this config file. This should remain unchanged

Reply:
Yes that's the idea. As you can see in that link you referenced, in a couple  of places the following is mentioned:
```
		// the identity is only required when the corresponding feature is enabled.
		// At the moment of writing, we don't have optional features that requires MI
		// once we start adding such features, an extra check needs to go here to
		// determine if the feature is disabled before adding the identity to the slices
		// of identities that are not needed
```

etcd data level encryption with kms is one of such features.